package.json

```diff
@@ -40,7 +41,7 @@
     "fs-extra": "10.0.0",
     "http-status-codes": "^2.1.4",
     "lodash": "^4.17.21",
-    "next": "^12.0.4",
+    "next": "12.0.4",
```

Comment:
What forces us to lock the version of `next`?

I don't know how related this is, but I get the following error when I `npm install`:

```
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: react-motion@0.5.2
npm WARN Found: react@17.0.2
npm WARN node_modules/react
npm WARN   react@"17.0.2" from the root project
npm WARN   34 more (@mdx-js/react, @next/react-dev-overlay, ...)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer react@"^0.14.9 || ^15.3.0 || ^16.0.0" from react-motion@0.5.2
npm WARN node_modules/@nivo/colors/node_modules/react-motion
npm WARN   react-motion@"^0.5.2" from @nivo/colors@0.78.0
npm WARN   node_modules/@nivo/colors
npm WARN 
npm WARN Conflicting peer dependency: react@16.14.0
npm WARN node_modules/react
npm WARN   peer react@"^0.14.9 || ^15.3.0 || ^16.0.0" from react-motion@0.5.2
npm WARN   node_modules/@nivo/colors/node_modules/react-motion
npm WARN     react-motion@"^0.5.2" from @nivo/colors@0.78.0
npm WARN     node_modules/@nivo/colors
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: 'postcss-nesting@8.0.1',
npm WARN EBADENGINE   required: { node: '12 - 16' },
npm WARN EBADENGINE   current: { node: 'v18.3.0', npm: '8.11.0' }
npm WARN EBADENGINE }
```

Reply:
> What forces us to lock the version of `next`?
> 
> I don't know how related this is, but I get the following error when I `npm install`:
> 
> ```
> npm WARN ERESOLVE overriding peer dependency
> npm WARN While resolving: react-motion@0.5.2
> npm WARN Found: react@17.0.2
> npm WARN node_modules/react
> npm WARN   react@"17.0.2" from the root project
> npm WARN   34 more (@mdx-js/react, @next/react-dev-overlay, ...)
> npm WARN 
> npm WARN Could not resolve dependency:
> npm WARN peer react@"^0.14.9 || ^15.3.0 || ^16.0.0" from react-motion@0.5.2
> npm WARN node_modules/@nivo/colors/node_modules/react-motion
> npm WARN   react-motion@"^0.5.2" from @nivo/colors@0.78.0
> npm WARN   node_modules/@nivo/colors
> npm WARN 
> npm WARN Conflicting peer dependency: react@16.14.0
> npm WARN node_modules/react
> npm WARN   peer react@"^0.14.9 || ^15.3.0 || ^16.0.0" from react-motion@0.5.2
> npm WARN   node_modules/@nivo/colors/node_modules/react-motion
> npm WARN     react-motion@"^0.5.2" from @nivo/colors@0.78.0
> npm WARN     node_modules/@nivo/colors
> npm WARN EBADENGINE Unsupported engine {
> npm WARN EBADENGINE   package: 'postcss-nesting@8.0.1',
> npm WARN EBADENGINE   required: { node: '12 - 16' },
> npm WARN EBADENGINE   current: { node: 'v18.3.0', npm: '8.11.0' }
> npm WARN EBADENGINE }
> ```

At the time of this PR, Next released a very problematic version that caused build to fail.
We fixed the version to prevent any problem.

Since then though, a patched version has been released https://github.com/vercel/next.js/releases/tag/v12.2.1.
This release (even though minor) still introduces a breaking change for the middleware functionnality (which was in beta) and that we use for i18n redirection.

Reply:
I tried to clone the repo and run `npm install`. I don't have the same feedback in the console which is very strange. @martinratinaud any ideas ?

```
❯ npm install
npm WARN old lockfile
npm WARN old lockfile The package-lock.json file was created with an old version of npm,
npm WARN old lockfile so supplemental metadata must be fetched from the registry.
npm WARN old lockfile
npm WARN old lockfile This is a one-time fix-up, please be patient...
npm WARN old lockfile
npm WARN deprecated @types/axios@0.14.0: This is a stub types definition for axios (https://github.com/mzabriskie/axios). axios provides its own type definitions, so you don't need @types/axios installed!
npm WARN deprecated @types/query-string@6.3.0: This is a stub types definition. query-string provides its own type definitions, so you do not need this installed.
npm WARN deprecated @types/react-icons@3.0.0: This is a stub types definition. react-icons provides its own type definitions, so you do not need this installed.
npm WARN deprecated stable@0.1.8: Modern JS already guarantees Array#sort() is a stable sort, so this library is deprecated. See the compatibility table on MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#browser_compatibility
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated svgo@1.3.2: This SVGO version is no longer supported. Upgrade to v2.x.x.
npm WARN deprecated tslint@6.1.3: TSLint has been deprecated in favor of ESLint. Please see https://github.com/palantir/tslint/issues/4534 for more information.
npm WARN deprecated core-js@2.6.12: core-js@<3.23.3 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Some versions have web compatibility issues. Please, upgrade your dependencies to the actual version of core-js.
npm WARN deprecated core-js@2.6.12: core-js@<3.23.3 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Some versions have web compatibility issues. Please, upgrade your dependencies to the actual version of core-js.
npm WARN deprecated core-js@3.19.1: core-js@<3.23.3 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Some versions have web compatibility issues. Please, upgrade your dependencies to the actual version of core-js.
```

Reply:
Ok done.